test/tests/amqp/test_amqp_racknode.py

```diff
@@ -0,0 +1,205 @@
+import threading
+import pika
+import logging
+import flogging
+import fit_common
+import test_api_utils
+from nose.plugins.attrib import attr
+
+amqp_message_received = False
+routing_key = ""
+amqp_body = ""
+logs = flogging.get_loggers()
+
+
+class AmqpWorker(threading.Thread):
+    '''
+    This AMQP worker Class will creat another thread when initialized and runs asynchronously.
+    The externalcallback is the callback function entrance for user.
+    The callback function will be call when AMQP message is received.
+    Each test case can define its own callback and pass the function name to the AMQP class.
+    The timeout parameter specify how long the AMQP daemon will run. self.panic is called when timeout.
+    eg:
+    def callback(self, ch, method, properties, body):
+        logs.debug(" [x] %r:%r" % (method.routing_key, body))
+        logs.debug(" [x] %r:%r" % (method.routing_key, body))
+
+    td = fit_amqp.AMQP_worker("node.added.#", callback)
+    td.setDaemon(True)
+    td.start()
+    '''
+
+    def __init__(self, exchange_name, topic_routing_key, external_callback, timeout=10):
+        threading.Thread.__init__(self)
+        pika_logger = logging.getLogger('pika')
+        if fit_common.VERBOSITY >= 8:
+            pika_logger.setLevel(logging.DEBUG)
+        elif fit_common.VERBOSITY >= 4:
+            pika_logger.setLevel(logging.WARNING)
+        else:
+            pika_logger.setLevel(logging.ERROR)
+        self.connection = pika.BlockingConnection(
+            pika.ConnectionParameters(host=fit_common.fitargs()["ora"], port=fit_common.fitports()['amqp']))
+        self.channel = self.connection.channel()
+        self.channel.basic_qos(prefetch_count=1)
+        result = self.channel.queue_declare(exclusive=True)
+        queue_name = result.method.queue
+        self.channel.queue_bind(
+            exchange=exchange_name,
+            queue=queue_name,
+            routing_key=topic_routing_key)
+        self.channel.basic_consume(external_callback, queue=queue_name)
+        self.connection.add_timeout(timeout, self.panic)
+
+    def panic(self):
+        self.channel.stop_consuming()
+        logs.debug_7('Pika connection timeout')
+        self.connection.close()
+        exit(0)
+
+    def run(self):
+        logs.debug_7('start consuming')
+        self.channel.start_consuming()
+
+
+@attr(all=True, regression=True, smoke=True)
+class amqp_rack_node(fit_common.unittest.TestCase):
+    # clear the test environment
+    def tear_down(self):
+        self.test_api_delete_rack()
+
+    def _compare_message(self, expected_key, expected_payload):
+        global routing_key, amqp_body
+        self.assertEquals(
+            routing_key, expected_key, "Routing key is not expected! expect {0}, get{1}"
+            .format(expected_key, routing_key))
+        try:
+            amqp_body_json = fit_common.json.loads(amqp_body)
+        except:
+            logs.error("FAILURE - The message body is not json format!")

```

Comment:
add the amqp body to this message please (it's likely misformatted json rather than not-json in any case)